lib/checkstl.cpp

```diff
@@ -2016,6 +2016,38 @@ static bool accumulateBool(const Token * tok, unsigned int varid)
     return false;
 }
 
+static bool hasVarIds(const Token * tok, unsigned int var1, unsigned int var2)
+{
+    if(tok->astOperand1()->varId() == tok->astOperand2()->varId())
+        return false;
+    unsigned int varids[] = {var1, var2};
+    unsigned int * varidsEnd = varids+2;
+    if(std::find(varids, varidsEnd, tok->astOperand1()->varId()) == varidsEnd)
```

Comment:
It is not wrong.. but I would probably not use `std::find` for 2 comparisons. A single return with 4 comparisons is enough.